python/ambassador/ir/irhealthchecks.py

```diff
@@ -0,0 +1,235 @@
+                "unhealthy_threshold": unhealthy_threshold,
+            }
+
+            # Process a http health check
+            if http_health_check is not None:
+                if not isinstance(http_health_check, dict):
+                    self.post_error(
+                        f"IRHealthChecks: http_health_check: field must be an object, found {http_health_check}. Ignoring health-check {hc}",
+                        log_level=logging.ERROR,
+                    )
+                    continue
+
+                # Make sure we have a path
+                path = http_health_check.get("path", None)
+                if path is None:
+                    self.post_error(
+                        f"IRHealthChecks: http_health_check.path is a required field. Ignoring health-check: {hc}",
+                        log_level=logging.ERROR,
+                    )
+                    continue
+                http_mapper: Dict[str, Any] = {"path": path}
+
+                # Process header add/remove operations
+                request_headers_to_add = http_health_check.get("add_request_headers", None)
+                if request_headers_to_add is not None:
+                    if isinstance(request_headers_to_add, list):
+                        self.post_error(
+                            f"IRHealthChecks: add_request_headers must be a dict of header:value pairs. Ignoring field for health-check: {hc}",
+                            log_level=logging.ERROR,
+                        )
+                    addHeaders = self.generate_headers_to_add(request_headers_to_add)
+                    if len(addHeaders) > 0:
+                        http_mapper["request_headers_to_add"] = addHeaders
+                request_headers_to_remove = http_health_check.get("remove_request_headers", None)
+                if request_headers_to_remove is not None:
+                    if not isinstance(request_headers_to_remove, list):
+                        self.post_error(
+                            f"IRHealthChecks: remove_request_headers must be a list. Ignoring field for health-check: {hc}",
+                            log_level=logging.ERROR,
+                        )
+                    else:
+                        http_mapper["request_headers_to_remove"] = request_headers_to_remove
+
+                host = http_health_check.get("hostname", None)
+                if host is not None:
+                    http_mapper["host"] = host
+
+                # Process the expected statuses
+                expected_statuses = http_health_check.get("expected_statuses", None)
+                if expected_statuses is not None:
+                    validStatuses = []
+                    for statusRange in expected_statuses:
+                        startCode = int(statusRange["start"])
+                        endCode = int(statusRange["end"])
+                        if startCode < 100 or startCode >= 600:
```

Comment:
This is unecessary isn't it? The CRD Validation should capture it on apply? Is there ever a time when this could occur short of calling diagd directly?

Reply:
Yeah, thats just me being paranoid that something would change in the future or if something else were to break. Something just kind of bothers me about trusting CRD Validation with some of these data types.

Reply:
Is 600 a valid code?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Reply:
FWIW the CRD validation will prevent you from applying one that has a value greater than or equal to `600` and the Envoy config is fine with `600` as a code so the main problems here are just that the increment of `end` should happen before the check and that there needs to be test coverage for this.

Reply:
@LanceEa 600 is not valid. "All valid status codes are within the range of 100 to 599, inclusive." https://www.rfc-editor.org/rfc/rfc9110#section-15-1

Reply:
@LukeShu - Yup, I miss-read the code above. I thought the code was allowing 600 to be a valid value. But after taking a look it was excluding 600 😄..doh